src/jit/emit.cpp

```diff
@@ -2969,9 +2999,10 @@ emitter::instrDesc  * emitter::emitNewInstrCallInd(int        argCnt,
                                                    VARSET_VALARG_TP GCvars,
                                                    regMaskTP  gcrefRegs,
                                                    regMaskTP  byrefRegs,
-                                                   emitAttr   retSizeIn)
+                                                   emitAttr   retSizeIn
+                                                   FEATURE_MULTIREG_RET_ONLY_ARG(emitAttr secondRetSizeIn))
 {
-    emitAttr  retSize = retSizeIn ? EA_ATTR(retSizeIn) : EA_PTRSIZE;
+    emitAttr       retSize = (retSizeIn != EA_UNKNOWN) ? retSizeIn : EA_PTRSIZE;
 
     bool gcRefRegsInScratch = ((gcrefRegs & RBM_CALLEE_TRASH) != 0);
```

Comment:
I thin @LLITCHEV is correct here.

The flow of gcrefRegs and byrefRegs is along the following call chain

genCallInstruction() --> 
genEmitCall() --> 
emitIns_call(gcrefRegs = gcinfo.gcRegGCrefSetCur, byrefRegs = gcinfo.gcRegByrefSetCur) --> emitNewInstrCallInd()

In fact emitIns_Call() removes any callee trash registers from gcrefRegs and byrefRegs before passing them to emitNewInstrCallInd.

While generating code for a call that returns a struct in two registers RAX/RDX,  those registers will not be in gcinfo.gcRegGCrefSetCur and gcinfo.gcRegByrefSetCur.  Hence while allocating InstDesc for such a call we would have to explicitly check whether the second return type is a gc-type.